rasa/server.py

```diff
@@ -564,19 +571,21 @@ async def append_events(request: Request, conversation_id: Text):
         try:
             async with app.agent.lock_store.lock(conversation_id):
                 processor = app.agent.create_processor()
-                tracker = processor.get_tracker(conversation_id)
-                output_channel = _get_output_channel(request, tracker)
-
                 events = _get_events_from_request_body(request)
 
-                for event in events:
-                    tracker.update(event, app.agent.domain)
+                tracker = await fetch_tracker_and_update_with_events(
```

Comment:
optional / nit picking: I'm not a huge fan of `and` in function names as a functions are supposed to do one thing and in this case the name seems a bit too focused on the `how` instead of the `what` . What about something like this? 
```suggestion
                tracker = await update_conversation_with_events(
```